sweep/sweeper.go

```diff
@@ -510,6 +510,82 @@ func (s *UtxoSweeper) feeRateForPreference(
 	return feeRate, nil
 }
 
+// removeLastSweepDescendants removes any transactions from the wallet that
+// spend outputs produced by the passed spendingTx. This needs to be done in
+// cases where we're not the only ones that can sweep an output, but there may
+// exists unconfirmed spends that spend outputs created by a sweep transaction.
+// The most common case for this is when someone sweeps our anchor outputs
+// after 16 blocks.
+func (s *UtxoSweeper) removeLastSweepDescendants(spendingTx *wire.MsgTx) error {
+	// Obtain all the past sweeps that we've done so far. We'll need these
+	// to ensure that if the spendingTx spends any of the same inputs, then
+	// we remove any transaction that may be spending those inputs from the
+	// wallet.
+	//
+	// TODO(roasbeef): can be last sweep here if we remove anything confirmed
+	// from the store?
+	pastSweepHashes, err := s.cfg.Store.ListSweeps()
```

Comment:
I wonder how large the sweeps can be? If a lot this may be expensive. 

> if we remove anything confirmed from the store

Do you think we can modify the `sweeper-last-tx` to store all unconfirmed txes? Atm it seems to only store the latest publish tx.

Reply:
Yeah that's the thing, we never delete the set of past sweeps as is. For very old nodes, this could end up being hundreds of past weeps. I chose to do this rather than just use the last sweep tx as it seems possible that we back off the failed sweep and then try another combination, causing us to not actually remove this set of conflicting transactions...

Though the disk footprint isn't very large (32 bytes per sweep), I think we should start to prune these sweeps once either we remove them here, we find a conflict, or they confirm with a block of depth N.

Reply:
So I guess for old sweeps we don't ever use them for anything...

Reply:
> So I guess for old sweeps we don't ever use them for anything...

Yeah nothing other than this new logic, but once the set of swept items are "well confirmed" then we no longer need them, I think it makes sense to roll out this garbage collection into a new issue, so we can expedite getting this into the next minor release.